### PR TITLE
Improve error handling of config file

### LINKED ISSUE #412: Settings in config file seem to be ignored
I have this config file:

```
network=regtest
datadir=/var/lib/loopd-system-regtest
rpclisten=0.0.0.0:11012
restlisten=0.0.0.0:11112
tlscertpath=/var/lib/loopd-system-regtest/loop.cert
tlskeypath=/var/lib/loopd-system-regtest/loop.key

lnd.host=127.0.0.1:10011

lnd.tlspath=/var/lib/lnd-system-regtest/public/tls.cert

lnd.macaroonpath=/var/lib/lnd-system-regtest/private/admin.macaroon
```

The `lnd.*` options seem to be ignored.

When I attempt to launch `loopd` 

**Expected behavior**

`loopd` reads certificate from `/var/lib/lnd-system-regtest/public/tls.cert`, macaron from `/var/lib/lnd-system-regtest/private/admin.macaroon` and connects to `127.0.0.1:10011`

**Actual behavior**

`loopd` attempts to connect to `127.0.0.1:10009` and read `/var/lib/loopd-system-regtest/.lnd/tls.cert`

Log output:
```
2021-08-03 14:29:21.840 [INF] LOOPD: Version: 0.14.2-beta commit=
2021-08-03 14:29:21.840 [INF] LNDC: Creating lnd connection to localhost:10009
loopd exited with an error: open /var/lib/loopd-system-regtest/.lnd/tls.cert: no such file or directory
```

**To reproduce**

Create the configuration file above and launch `loopd` with argument `--configfile /etc/loopd-system-regtest/loopd.conf`

**System information**

* Standalone binary
* Interaction with loop impossible because it can't start
* Loopd version: 0.14.2 - downloaded from releases
* Machine: Debian 10, x86_64, NVMe SSD
* OS: Debian 10

### PR DESCRIPTION
**Pull request by Kixunil**
This makes two changes to handling of errors when configuration file
could not be loaded:

1. Only NotFound errors are considered OK - access errors and other FS
   issues are now treated as fatal.
2. Failing to load config file specified explicitly via `--configfile`
   option is alway a fatal error.

Rationale: If the configfile was specified explicitly then it indicates
the user really wishes to load it. While the user could want it to be
optionally loaded for extra configuration options, this can be
accomplished using an empty file.

If the config file was not specified explicitly then its' path was
computed from loop directory. If the file is inaccessible due to
permissions or other FS errors it's nearly certain other following
operations will fail as well. Failing early with a clear message is thus
beneficial.

This still leaves room for uncaught user error (e.g. mistakenly naming
config file inside loop dir as `loop.conf` instead of `loopd.conf`) but
it's greatly reduced and such error should be easier to identify.

(Indirectly) closes #412

#### Pull Request Checklist
- [x] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

File: release_notes.md [hunk 1]
```diff
@@ -18,6 +18,12 @@ This file tracks release notes for the loop client.
 
 #### Breaking Changes
 
+* Failing to load configuration file specified by `--configfile` for any reason if now hard error.
```
Comment (carlaKC) [inline release_notes.md:21]: nit: wrap at 80

File: loopd/run.go [hunk 2]
```diff
@@ -149,13 +149,16 @@ func Run(rpcCfg RPCConfig) error {
 
 	// Parse ini file.
 	loopDir := lncfg.CleanAndExpandPath(config.LoopDir)
-	configFile := getConfigPath(config, loopDir)
+	configFile, explicitConfig := getConfigPath(config, loopDir)
```
Comment (carlaKC) [inline loopd/run.go:152]: nit: `customConfig` makes a little more sense than `explicitConfig` to me, but nonblocking

**Review by carlaKC — APPROVED** [review]
tACK! Thank you for the fix, great improvement :)

File: release_notes.md (hunk 1 above)
Comment (Kixunil) [reply release_notes.md:21]: Hmm, previous lines weren't 80-wrapped so I thought the limit doesn't apply. Will fix.

File: loopd/run.go (hunk 2 above)
Comment (Kixunil) [reply loopd/run.go:152]: I'm not really sure. Each config is custom. :) Maybe `nonDefaultConfigPath`?

File: release_notes.md (hunk 1 above)
Comment (Kixunil) [reply release_notes.md]: Done.

File: loopd/run.go (hunk 2 above)
Comment (yyforyongyu) [reply loopd/run.go:152]: maybe an action in the name? something like `hasExplicitConfig`,`hasConfigSet`, etc...

**Review by yyforyongyu — APPROVED** [review]
Thanks for the PR @Kixunil ! Changes look good to me, just a few nits.

**Comment by Kixunil** [comment]
Addressed the nits.

**Comment by carlaKC** [comment]
Thanks for this one, great improvement! 

Tested: 
1. `--configfile` exists -> ok
2. `--configfile` doesn't exist -> exits
3. No `--configfile`, default location cfg file does not exist -> ok

**Pull request merged**